src/components/Slider/Slider.js

```diff
@@ -1,92 +1,99 @@
-import React from 'react';
-import omit from 'lodash/omit';
+import times from 'lodash/times';
+import range from 'lodash/range';
+import has from 'lodash/has';
 
-import Nouislider from 'react-nouislider';
+import React, {Component, PropTypes} from 'react';
+import Rheostat from 'rheostat';
+import cx from 'classnames';
 
-const cssPrefix = 'ais-range-slider--';
-
-import isEqual from 'lodash/isEqual';
+import Pit from './Pit.js';
 
 import autoHideContainerHOC from '../../decorators/autoHideContainer.js';
 import headerFooterHOC from '../../decorators/headerFooter.js';
 
-export class RawSlider extends React.Component {
-  componentWillMount() {
-    this.handleChange = this.handleChange.bind(this);
+class Slider extends Component {
+
+  static propTypes = {
+    refine: PropTypes.func.isRequired,
+    min: PropTypes.number.isRequired,
+    max: PropTypes.number.isRequired,
+    values: PropTypes.arrayOf(PropTypes.number).isRequired,
+    pips: PropTypes.oneOfType([
+      PropTypes.bool,
+      PropTypes.object,
+    ]),
+    step: PropTypes.number.isRequired,
+    tooltips: PropTypes.oneOfType([
+      PropTypes.bool,
+      PropTypes.shape({format: PropTypes.func.isRequired}),
+    ]),
   }
 
-  shouldComponentUpdate(nextProps) {
-    return !isEqual(this.props.range, nextProps.range) ||
-      !isEqual(this.props.start, nextProps.start);
+  handleChange = ({values}) => {
+    const {refine} = this.props;
+    refine(values);
   }
 
-  // we are only interested in rawValues
-  handleChange(formattedValues, handleId, rawValues) {
-    this.props.onChange(rawValues);
+  // creates an array number where to display a pit point on the slider
+  computeDefaultPitPoints({min, max}) {
+    const totalLenght = max - min;
+    const steps = 34;
+    const stepsLength = totalLenght / steps;
+
+    const pitPoints = [min, ...times(steps - 1, step => min + stepsLength * (step + 1)), max]
+      // bug with `key={ 0 }` and preact, see https://github.com/developit/preact/issues/642
+      .map(pitPoint => pitPoint === 0 ? 0.00001 : pitPoint);
+
+    return pitPoints;
+  }
+
+  // creates an array of values where the slider should snap to
+  computeSnapPoints({min, max, step}) {
+    const r = range(min, max, step);
+    return r.includes(max) ? r : [...r.slice(0, -1), max];
```

Comment:
Range never includes max, https://lodash.com/docs/4.17.4#range Not sure what we're trying to achieve, but there might be an edge case if the last element of `r` is `max - step`.

Also mutating r would be more conservative. `r.slice(0, -1).push(max)` or even using `splice`.